File: BPMOPMobile_VietNamAirLines/BPMOPMobile_VietNamAirLines/BPMOPMobile.Droid/Presenter/SharedView/SharedView_PopupActionAccept.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.Graphics;
using Android.Graphics.Drawables;
using Android.OS;
using Android.Runtime;
using Android.Support.V4.Content.Res;
using Android.Support.V7.App;
using Android.Views;
using Android.Widget;
using BPMOPMobile.Bean;
using BPMOPMobile.Class;
using BPMOPMobile.Droid.Class;
using BPMOPMobile.Droid.Core.Common;
using BPMOPMobile.Droid.Presenter.Fragment;

namespace BPMOPMobile.Droid.Presenter.SharedView
{
    public class SharedView_PopupActionAccept : SharedView_PopupActionBase
    {
        public SharedView_PopupActionAccept(LayoutInflater _inflater, AppCompatActivity _mainAct, CustomBaseFragment _fragment, string _fragmentTag, View _rootView) : base(_inflater, _mainAct, _fragment, _fragmentTag, _rootView)
        {
        }

        public override void InitializeValue_DetailWorkflow(ButtonAction buttonAction)
        {
            base.InitializeValue_DetailWorkflow(buttonAction);
        }

        public override void InitializeView()
        {
            base.InitializeView();
            try
            {
                #region Get View - Init Data
                View _viewPopupAction = _inflater.Inflate(Resource.Layout.PopupAction_Accept, null);
                TextView _tvTitle = _viewPopupAction.FindViewById<TextView>(Resource.Id.tv_PopupAction_Accept_Title);
                ImageView _imgAction = _viewPopupAction.FindViewById<ImageView>(Resource.Id.img_PopupAction_Accept);
                EditText _edtComment = _viewPopupAction.FindViewById<EditText>(Resource.Id.edt_PopupAction_Accept_YKien);
                TextView _tvCancel = _viewPopupAction.FindViewById<TextView>(Resource.Id.tv_PopupAction_Accept_Huy);
                TextView _tvAccept = _viewPopupAction.FindViewById<TextView>(Resource.Id.tv_PopupAction_Accept_HoanTat);
                TextView _tvNote = _viewPopupAction.FindViewById<TextView>(Resource.Id.tv_PopupAction_Accept_Note);

                _tvTitle.Text = buttonAction.Title;
                _edtComment.Hint = CmmFunction.GetTitle("MESS_REQUIRE_COMMENT", "Vui lòng nhập ý kiến");
                _tvCancel.Text = CmmFunction.GetTitle("TEXT_EXIT", "Thoát");
                _tvAccept.Text = buttonAction.Title;

                string _imageName = "icon_bpm_Btn_action_" + buttonAction.ID.ToString();
                int resId = _mainAct.Resources.GetIdentifier(_imageName.ToLowerInvariant(), "drawable", _mainAct.PackageName);
                _imgAction.SetImageResource(resId);

                if (buttonAction.Notes != null && buttonAction.Notes.Count > 0) // Check xem phiếu có đang tham vấn ko
                {
                    _tvNote.Text = buttonAction.Notes[0].Value;
                }

                #endregion

                #region Show View
                Dialog _dialogAction = new Dialog(_rootView.Context);
                Window window = _dialogAction.Window;
                _dialogAction.RequestWindowFeature(1);
                _dialogAction.SetCanceledOnTouchOutside(false);
                _dialogAction.SetCancelable(true);
                window.SetSoftInputMode(SoftInput.AdjustPan); // Để Hide soft keyboard ko bị Resize
                window.SetGravity(GravityFlags.Center);
                var dm = _mainAct.Resources.DisplayMetrics;

                _dialogAction.SetContentView(_viewPopupAction);
                _dialogAction.Show();
                WindowManagerLayoutParams s = window.Attributes;
                s.Width = dm.WidthPixels;
                window.Attributes = s;
                window.SetBackgroundDrawable(new ColorDrawable(Color.Transparent));
                #endregion

                #region Event
                _tvCancel.Click += (sender, e) =>
                {
                    CmmDroidFunction.HideSoftKeyBoard(_edtComment, _mainAct);
                    _dialogAction.Dismiss();
                };

                _edtComment.TextChanged += (sender, e) =>
                {
                    if (String.IsNullOrEmpty(_edtComment.Text))
                        _edtComment.SetTypeface(ResourcesCompat.GetFont(_mainAct, Resource.Font.fontarial), TypefaceStyle.Italic);
                    else
                        _edtComment.SetTypeface(ResourcesCompat.GetFont(_mainAct, Resource.Font.fontarial), TypefaceStyle.Normal);
                };

                _tvAccept.Click += (sender, e) =>
                {
                    CmmDroidFunction.HideSoftKeyBoard(_edtComment, _mainAct);
                    _dialogAction.Dismiss();

                    switch (flagView)
                    {
                        case (int)FlagViewControlAction.DetailWorkflow: // Action Trang Detail Workflow
                            {
                                FragmentDetailWorkflow _frag = (FragmentDetailWorkflow)_fragment;
                                _frag.Action_SendAPI(buttonAction, !String.IsNullOrEmpty(_edtComment.Text) ? _edtComment.Text : "", null); // Check xem có comment không
                                break;
                            }
                    }
                };
                #endregion

                _edtComment.Text = "";
            }
            catch (Exception ex)
            {
#if DEBUG
                CmmDroidFunction.WriteTrackingError(this.GetType().Name, "InitializeView", ex);
#endif
            }
        }
    }
}

File: VNABPMOP/app/src/main/java/com/vuthao/bpmop/shareview/SharedView_PopupActionAccept.java
package com.vuthao.bpmop.shareview;

import android.app.Activity;
import android.app.Dialog;
import android.content.Intent;
import android.graphics.Color;
import android.graphics.Typeface;
import android.graphics.drawable.ColorDrawable;
import android.text.Editable;
import android.text.TextWatcher;
import android.util.DisplayMetrics;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.core.content.res.ResourcesCompat;
import androidx.localbroadcastmanager.content.LocalBroadcastManager;

import com.google.gson.Gson;
import com.vuthao.bpmop.R;
import com.vuthao.bpmop.base.BroadcastUtility;
import com.vuthao.bpmop.base.Functions;
import com.vuthao.bpmop.base.keyboard.KeyboardManager;
import com.vuthao.bpmop.base.model.custom.ButtonAction;
import com.vuthao.bpmop.base.vars.VarsReceiver;
import com.vuthao.bpmop.core.Vars;

public class SharedView_PopupActionAccept extends SharedView_PopupActionBase implements TextWatcher, View.OnClickListener {
    private Dialog dialog;
    private View view;
    private TextView tvTitle;
    private ImageView imgAction;
    private EditText edtComment;
    private TextView tvCancel;
    private TextView tvAccept;
    private TextView tvNote;

    public SharedView_PopupActionAccept(LayoutInflater inflater, Activity mainAct, String fragmentTag, View rootView) {
        super(inflater, mainAct, fragmentTag, rootView);
    }

    @Override
    public void initializeValue_DetailWorkflow(ButtonAction buttonAction) {
        super.initializeValue_DetailWorkflow(buttonAction);
    }

    @Override
    public void initializeView() {
        super.initializeView();

        view = inflater.inflate(R.layout.popup_action_accept, null);
        tvTitle = view.findViewById(R.id.tv_PopupAction_Accept_Title);
        imgAction = view.findViewById(R.id.img_PopupAction_Accept);
        edtComment = view.findViewById(R.id.edt_PopupAction_Accept_YKien);
        tvCancel = view.findViewById(R.id.tv_PopupAction_Accept_Huy);
        tvAccept = view.findViewById(R.id.tv_PopupAction_Accept_HoanTat);
        tvNote = view.findViewById(R.id.tv_PopupAction_Accept_Note);

        tvTitle.setText(buttonAction.getTitle());
        edtComment.setHint(Functions.share.getTitle("MESS_REQUIRE_COMMENT", "Vui l??ng nh???p ?? ki???n"));
        tvCancel.setText(Functions.share.getTitle("TEXT_EXIT", "Tho??t"));
        tvAccept.setText(buttonAction.getTitle());

        String imageName = "icon_bpm_Btn_action_" + buttonAction.getID();
        int resId = mainAct.getResources().getIdentifier(imageName.toLowerCase(), "drawable", mainAct.getPackageName());
        imgAction.setImageResource(resId);

        if (buttonAction.getNotes() != null && buttonAction.getNotes().size() > 0) {
            tvNote.setText(buttonAction.getNotes().get(0).getValue());
        }

        // region Dialog
        dialog = new Dialog(mainAct);
        Window window = dialog.getWindow();
        dialog.requestWindowFeature(1);
        dialog.setCancelable(true);
        dialog.setCanceledOnTouchOutside(false);
        window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN);
        window.setGravity(Gravity.CENTER);

        DisplayMetrics displayMetrics = mainAct.getResources().getDisplayMetrics();
        dialog.setContentView(view);
        dialog.show();

        WindowManager.LayoutParams s = window.getAttributes();
        s.width = displayMetrics.widthPixels;
        window.setAttributes(s);
        window.setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        // endregion

        tvCancel.setOnClickListener(this);
        edtComment.addTextChangedListener(this);
        tvAccept.setOnClickListener(this);
        edtComment.setText("");
    }

    @Override
    public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

    }

    @Override
    public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {

    }

    @Override
    public void afterTextChanged(Editable s) {
        if (s.length() > 0) {
            edtComment.setTypeface(ResourcesCompat.getFont(mainAct, R.font.fontarial), Typeface.ITALIC);
        } else {
            edtComment.setTypeface(ResourcesCompat.getFont(mainAct, R.font.fontarial), Typeface.NORMAL);
        }
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.tv_PopupAction_Accept_Huy: {
                KeyboardManager.hideKeyboard(edtComment, mainAct);
                dialog.dismiss();
                break;
            }
            case R.id.tv_PopupAction_Accept_HoanTat: {
                Intent intent = new Intent();
                intent.setAction(VarsReceiver.SUBMITACTION);
                intent.putExtra("buttonAction", new Gson().toJson(buttonAction));
                intent.putExtra("comment", edtComment.getText().toString());
                BroadcastUtility.send(mainAct, intent);
                break;
            }
        }
    }
}
